### DISCUSSION TITLE
Ignoring warning in fixtures

### DISCUSSION THREAD

**Discussion by PerchunPak** [pytest-dev/pytest#10000] [Q&A] [answered]
Only useful what I found in internet - [this unanswered stackoverflow thread](https://stackoverflow.com/questions/55305362/pytest-ignore-warning-in-fixture). So is there any way to handle warnings in fixture?

P.S. I found also that `pytestmark = pytest.mark.filterwarnings("error")` doesn't change anything (there is a couple of warning more raised later in this file).

---

Actually problem that my `deprecated` decorator turning off warnings capture.
```py
warnings.simplefilter("always", DeprecationWarning)
warnings.warn(warn_message, category=DeprecationWarning)
warnings.simplefilter("default", DeprecationWarning)
```

**Comment by The-Compiler** [comment, marked as answer]
You should be able to use normal Python mechanisms to suppress the warnings: https://docs.python.org/3/library/warnings.html#temporarily-suppressing-warnings
